substrate/dex/pallet/src/lib.rs

```diff
@@ -154,11 +154,34 @@ pub mod pallet {
   #[pallet::storage]
   pub type Pools<T: Config> = StorageMap<_, Blake2_128Concat, PoolId, PoolInfo<Coin>, OptionQuery>;
 
+  #[pallet::storage]
+  #[pallet::getter(fn last_quote_for_block)]
+  pub type LastQuoteForBlock<T: Config> =
+    StorageDoubleMap<_, Identity, BlockNumberFor<T>, Identity, Coin, [u8; 8], ValueQuery>;
+
+  /// Moving window of oracle prices.
+  #[pallet::storage]
+  #[pallet::getter(fn oracle_prices)]
+  pub type OraclePrices<T: Config> = StorageMap<_, Identity, [u8; 8], u16, OptionQuery>;
+  impl<T: Config> Pallet<T> {
+    /// Get the highest sustained(lowest) value of the window
+    pub fn highest_sustained_price() -> Amount {
+      let mut iter = OraclePrices::<T>::iter_keys();
+      // we can unwrap because we will always have at least 1 key in this map
+      // before this function is called.
+      Amount(u64::from_be_bytes(iter.next().unwrap()))
```

Comment:
I'd add a note this was the lowest price due to the keys being lexicographically ordered.